<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Universe of Goo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="GLPanel" name="GLWidget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>731</width>
      <height>731</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
   </widget>
   <widget class="QFrame" name="parameterFrame">
    <property name="geometry">
     <rect>
      <x>749</x>
      <y>-1</y>
      <width>441</width>
      <height>731</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>9</x>
       <y>-1</y>
       <width>431</width>
       <height>731</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="simOptionsBox">
        <property name="title">
         <string>Simulation Options</string>
        </property>
        <widget class="QWidget" name="horizontalLayoutWidget">
         <property name="geometry">
          <rect>
           <x>9</x>
           <y>19</y>
           <width>421</width>
           <height>211</height>
          </rect>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QGroupBox" name="ConstraintBox">
            <property name="title">
             <string>Constraint Handling</string>
            </property>
            <widget class="QRadioButton" name="penaltyForceButton">
             <property name="geometry">
              <rect>
               <x>20</x>
               <y>30</y>
               <width>121</width>
               <height>22</height>
              </rect>
             </property>
             <property name="text">
              <string>Penalty Force</string>
             </property>
            </widget>
            <widget class="QRadioButton" name="stepAndProjectButton">
             <property name="geometry">
              <rect>
               <x>20</x>
               <y>50</y>
               <width>181</width>
               <height>22</height>
              </rect>
             </property>
             <property name="text">
              <string>Step and Project</string>
             </property>
            </widget>
            <widget class="QRadioButton" name="lagranceMultiplierButton">
             <property name="geometry">
              <rect>
               <x>20</x>
               <y>70</y>
               <width>181</width>
               <height>22</height>
              </rect>
             </property>
             <property name="text">
              <string>Lagrange Multiplier</string>
             </property>
            </widget>
            <widget class="QPushButton" name="startSimulationButton">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>130</y>
               <width>181</width>
               <height>27</height>
              </rect>
             </property>
             <property name="text">
              <string>Start Simulation</string>
             </property>
            </widget>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="SimParametersBox">
            <property name="title">
             <string>Parameters</string>
            </property>
            <widget class="QLabel" name="timeStepLabel">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>30</y>
               <width>81</width>
               <height>21</height>
              </rect>
             </property>
             <property name="text">
              <string>Time Step:</string>
             </property>
            </widget>
            <widget class="QLabel" name="newtonTolLabel">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>50</y>
               <width>131</width>
               <height>21</height>
              </rect>
             </property>
             <property name="text">
              <string>Newton Tolerance:</string>
             </property>
            </widget>
            <widget class="QLabel" name="newtonMaxItersLabel">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>70</y>
               <width>131</width>
               <height>21</height>
              </rect>
             </property>
             <property name="text">
              <string>Newton Max Iters:</string>
             </property>
            </widget>
            <widget class="QLineEdit" name="timeStepEdit">
             <property name="geometry">
              <rect>
               <x>150</x>
               <y>30</y>
               <width>51</width>
               <height>21</height>
              </rect>
             </property>
            </widget>
            <widget class="QLineEdit" name="newtonTolEdit">
             <property name="geometry">
              <rect>
               <x>150</x>
               <y>50</y>
               <width>51</width>
               <height>21</height>
              </rect>
             </property>
            </widget>
            <widget class="QLineEdit" name="newtonMaxItersEdit">
             <property name="geometry">
              <rect>
               <x>150</x>
               <y>70</y>
               <width>51</width>
               <height>21</height>
              </rect>
             </property>
            </widget>
            <widget class="QLineEdit" name="penaltyStiffnessEdit">
             <property name="geometry">
              <rect>
               <x>150</x>
               <y>90</y>
               <width>51</width>
               <height>21</height>
              </rect>
             </property>
            </widget>
            <widget class="QLabel" name="penaltyStiffness">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>90</y>
               <width>131</width>
               <height>21</height>
              </rect>
             </property>
             <property name="text">
              <string>Penalty Stiffness:</string>
             </property>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="activeForcesBox">
        <property name="title">
         <string>Active Forces</string>
        </property>
        <widget class="QCheckBox" name="gravityCheckBox">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>30</y>
           <width>97</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Gravity</string>
         </property>
        </widget>
        <widget class="QCheckBox" name="springsCheckBox">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>50</y>
           <width>97</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Springs</string>
         </property>
        </widget>
        <widget class="QCheckBox" name="floorCheckBox">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>90</y>
           <width>97</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Floor</string>
         </property>
        </widget>
        <widget class="QLabel" name="gravityGLabel">
         <property name="geometry">
          <rect>
           <x>230</x>
           <y>30</y>
           <width>121</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Acceleration:</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="gravityGEdit">
         <property name="geometry">
          <rect>
           <x>370</x>
           <y>30</y>
           <width>51</width>
           <height>21</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="springStiffnessLabel">
         <property name="geometry">
          <rect>
           <x>230</x>
           <y>50</y>
           <width>121</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Base Stiffness:</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="springStiffnessEdit">
         <property name="geometry">
          <rect>
           <x>370</x>
           <y>50</y>
           <width>51</width>
           <height>21</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="maxStrainLabel">
         <property name="geometry">
          <rect>
           <x>230</x>
           <y>70</y>
           <width>121</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Max Strain:</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="maxStrainEdit">
         <property name="geometry">
          <rect>
           <x>370</x>
           <y>70</y>
           <width>51</width>
           <height>21</height>
          </rect>
         </property>
        </widget>
        <widget class="QCheckBox" name="dampingStiffnessCheckBox">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>110</y>
           <width>151</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Viscous Damping</string>
         </property>
        </widget>
        <widget class="QLabel" name="dampingStiffnessLabel">
         <property name="geometry">
          <rect>
           <x>230</x>
           <y>110</y>
           <width>121</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Stiffness:</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="dampingStiffnessEdit">
         <property name="geometry">
          <rect>
           <x>370</x>
           <y>110</y>
           <width>51</width>
           <height>21</height>
          </rect>
         </property>
        </widget>
        <widget class="QCheckBox" name="elasticBendingCheckbox">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>130</y>
           <width>151</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Elastic Bending</string>
         </property>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="uiOptionsBox">
        <property name="title">
         <string>UI Options</string>
        </property>
        <widget class="QWidget" name="horizontalLayoutWidget_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>421</width>
           <height>211</height>
          </rect>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QGroupBox" name="clickFunctionBox">
            <property name="title">
             <string>Click Function</string>
            </property>
            <widget class="QRadioButton" name="addParticleButton">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>30</y>
               <width>117</width>
               <height>21</height>
              </rect>
             </property>
             <property name="text">
              <string>Add Particle</string>
             </property>
            </widget>
            <widget class="QRadioButton" name="addSawButton">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>50</y>
               <width>117</width>
               <height>21</height>
              </rect>
             </property>
             <property name="text">
              <string>Add Saw</string>
             </property>
            </widget>
            <widget class="QCheckBox" name="isFixedCheckBox">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>70</y>
               <width>97</width>
               <height>21</height>
              </rect>
             </property>
             <property name="text">
              <string>Is Fixed</string>
             </property>
            </widget>
            <widget class="QGroupBox" name="ConnectorTypeBox">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>90</y>
               <width>136</width>
               <height>209</height>
              </rect>
             </property>
             <property name="title">
              <string>Connector Type</string>
             </property>
             <widget class="QRadioButton" name="springButton">
              <property name="geometry">
               <rect>
                <x>10</x>
                <y>30</y>
                <width>117</width>
                <height>21</height>
               </rect>
              </property>
              <property name="text">
               <string>Spring</string>
              </property>
             </widget>
             <widget class="QRadioButton" name="flexibleRodButton">
              <property name="geometry">
               <rect>
                <x>10</x>
                <y>70</y>
                <width>117</width>
                <height>21</height>
               </rect>
              </property>
              <property name="text">
               <string>FlexibleRod</string>
              </property>
             </widget>
             <widget class="QRadioButton" name="ropeButton">
              <property name="geometry">
               <rect>
                <x>10</x>
                <y>90</y>
                <width>117</width>
                <height>21</height>
               </rect>
              </property>
              <property name="text">
               <string>Rope</string>
              </property>
             </widget>
             <widget class="QRadioButton" name="rigidRodButton">
              <property name="geometry">
               <rect>
                <x>10</x>
                <y>50</y>
                <width>117</width>
                <height>21</height>
               </rect>
              </property>
              <property name="text">
               <string>Rigid Rod</string>
              </property>
             </widget>
            </widget>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="particleSettingsBox">
            <property name="title">
             <string>Particle Settings</string>
            </property>
            <widget class="QLabel" name="massLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>30</y>
               <width>41</width>
               <height>21</height>
              </rect>
             </property>
             <property name="text">
              <string>Mass:</string>
             </property>
            </widget>
            <widget class="QLabel" name="maxSpringDistLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>50</y>
               <width>111</width>
               <height>21</height>
              </rect>
             </property>
             <property name="text">
              <string>Max Spring Dist:</string>
             </property>
            </widget>
            <widget class="QLineEdit" name="massEdit">
             <property name="geometry">
              <rect>
               <x>70</x>
               <y>30</y>
               <width>51</width>
               <height>21</height>
              </rect>
             </property>
            </widget>
            <widget class="QLineEdit" name="maxSpringDistEdit">
             <property name="geometry">
              <rect>
               <x>70</x>
               <y>70</y>
               <width>51</width>
               <height>21</height>
              </rect>
             </property>
            </widget>
            <widget class="QGroupBox" name="rodSettingsBox">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>90</y>
               <width>135</width>
               <height>209</height>
              </rect>
             </property>
             <property name="title">
              <string>Rod Settings</string>
             </property>
             <widget class="QLabel" name="rodDensityLabel">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>30</y>
                <width>61</width>
                <height>21</height>
               </rect>
              </property>
              <property name="text">
               <string>Density:</string>
              </property>
             </widget>
             <widget class="QLabel" name="rodStretchLabel">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>50</y>
                <width>61</width>
                <height>21</height>
               </rect>
              </property>
              <property name="text">
               <string>Stretch k</string>
              </property>
             </widget>
             <widget class="QLineEdit" name="rodDensityEdit">
              <property name="geometry">
               <rect>
                <x>70</x>
                <y>30</y>
                <width>51</width>
                <height>21</height>
               </rect>
              </property>
             </widget>
             <widget class="QLineEdit" name="rodStretchEdit">
              <property name="geometry">
               <rect>
                <x>70</x>
                <y>50</y>
                <width>51</width>
                <height>21</height>
               </rect>
              </property>
             </widget>
             <widget class="QLabel" name="rodBendLabel">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>70</y>
                <width>61</width>
                <height>21</height>
               </rect>
              </property>
              <property name="text">
               <string>Bend k</string>
              </property>
             </widget>
             <widget class="QLineEdit" name="rodBendEdit">
              <property name="geometry">
               <rect>
                <x>70</x>
                <y>70</y>
                <width>51</width>
                <height>21</height>
               </rect>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
             <widget class="QLineEdit" name="rodSegmentsEdit">
              <property name="geometry">
               <rect>
                <x>70</x>
                <y>90</y>
                <width>51</width>
                <height>21</height>
               </rect>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
             <widget class="QLabel" name="rodSegmentsLabel">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>90</y>
                <width>61</width>
                <height>21</height>
               </rect>
              </property>
              <property name="text">
               <string>Segments</string>
              </property>
             </widget>
            </widget>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="sawSettingsBox">
            <property name="title">
             <string>Saw Settings</string>
            </property>
            <widget class="QLabel" name="radiusLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>30</y>
               <width>51</width>
               <height>21</height>
              </rect>
             </property>
             <property name="text">
              <string>Radius:</string>
             </property>
            </widget>
            <widget class="QLineEdit" name="radiusEdit">
             <property name="geometry">
              <rect>
               <x>80</x>
               <y>30</y>
               <width>51</width>
               <height>21</height>
              </rect>
             </property>
            </widget>
            <widget class="QGroupBox" name="ropeSettingsBox">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>90</y>
               <width>136</width>
               <height>209</height>
              </rect>
             </property>
             <property name="title">
              <string>Rope Settings</string>
             </property>
             <widget class="QLabel" name="ropeDensityLabel">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>30</y>
                <width>51</width>
                <height>21</height>
               </rect>
              </property>
              <property name="text">
               <string>Density:</string>
              </property>
             </widget>
             <widget class="QLineEdit" name="ropeDensityEdit">
              <property name="geometry">
               <rect>
                <x>80</x>
                <y>30</y>
                <width>51</width>
                <height>21</height>
               </rect>
              </property>
             </widget>
             <widget class="QLabel" name="ropeBendkLabel">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>50</y>
                <width>51</width>
                <height>21</height>
               </rect>
              </property>
              <property name="text">
               <string>Bend k</string>
              </property>
             </widget>
             <widget class="QLineEdit" name="ropeBendEdit">
              <property name="geometry">
               <rect>
                <x>80</x>
                <y>50</y>
                <width>51</width>
                <height>21</height>
               </rect>
              </property>
             </widget>
             <widget class="QLabel" name="ropeSegmentsLabel">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>70</y>
                <width>81</width>
                <height>21</height>
               </rect>
              </property>
              <property name="text">
               <string>Segments</string>
              </property>
             </widget>
             <widget class="QLineEdit" name="ropeSegmentsEdit">
              <property name="geometry">
               <rect>
                <x>80</x>
                <y>70</y>
                <width>51</width>
                <height>21</height>
               </rect>
              </property>
             </widget>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuScene">
    <property name="title">
     <string>Scene</string>
    </property>
    <addaction name="actionReset"/>
    <addaction name="actionReset_Everything"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuScene"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="text">
    <string>Clear Scene</string>
   </property>
  </action>
  <action name="actionReset_Everything">
   <property name="text">
    <string>Reset Everything</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLPanel</class>
   <extends>QWidget</extends>
   <header>glpanel.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
